src/terminal/parser/stateMachine.cpp

```diff
@@ -1337,64 +1235,56 @@ bool StateMachine::FlushToTerminal()
 //     a escape sequence, then feed characters into the state machine one at a
 //     time until we return to the ground state.
 // Arguments:
-// - rgwch - Array of new characters to operate upon
-// - cch - Count of characters in array
+// - string - Characters to operate upon
 // Return Value:
 // - <none>
-void StateMachine::ProcessString(const wchar_t* const rgwch, const size_t cch)
+void StateMachine::ProcessString(const std::wstring_view string)
 {
-    _pwchCurr = rgwch;
-    _pwchSequenceStart = rgwch;
-    _currRunLength = 0;
+    size_t start = 0;
+    size_t current = start;
 
-    for (size_t cchCharsRemaining = cch; cchCharsRemaining > 0; cchCharsRemaining--)
+    while (current < string.size())
     {
-        if (_fProcessingIndividually)
+        _run = string.substr(start, current - start);
+
+        if (_processingIndividually)
         {
             // If we're processing characters individually, send it to the state machine.
-            ProcessCharacter(*_pwchCurr);
-            _pwchCurr++;
+            ProcessCharacter(string.at(current));

```

Comment:
is this a candidate for the `prechecked_at` we discussed? current _is_ `<` string.size

Reply:
if it is, should it be an `operator[]`?

Reply:
Yes but in the next PR with audit.

Reply:
I've noted it to carry over there.